tests/test_buffer_dynamic.py

```diff
@@ -39,12 +51,65 @@ def setup_db(self, dvs):
-        return "pg_lossless_" + speed + "_" + cable_length + "_profile"
+    def setup_asic_db(self, dvs):
+        buffer_pool_set = set(self.asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_POOL"))
+        self.initProfileSet = set(self.asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE"))
+        self.initPGSet = set(self.asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP"))
+
+        ingress_lossless_pool = self.app_db.get_entry("BUFFER_POOL_TABLE", "ingress_lossless_pool")
+        self.ingress_lossless_pool_asic = None
+
+        for key in buffer_pool_set:
+            bufferpool = self.asic_db.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_POOL", key)
+            if bufferpool["SAI_BUFFER_POOL_ATTR_TYPE"] == "SAI_BUFFER_POOL_TYPE_INGRESS":
+                if ingress_lossless_pool["size"] == bufferpool["SAI_BUFFER_POOL_ATTR_SIZE"]:
+                    self.ingress_lossless_pool_asic = bufferpool
+                    self.ingress_lossless_pool_oid = key
+
+    def check_new_profile_in_asic_db(self, dvs, profile):
+        diff = set(self.asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE")) - self.initProfileSet
+        if len(diff) == 1:
+            self.newProfileInAsicDb = diff.pop()
+        # in case diff is empty, we just treat the newProfileInAsicDb cached the latest one
+        fvs = self.app_db.get_entry("BUFFER_PROFILE_TABLE", profile)
+        if fvs.get('dynamic_th'):
+            sai_threshold_value = fvs['dynamic_th']
+            sai_threshold_mode = 'SAI_BUFFER_PROFILE_THRESHOLD_MODE_DYNAMIC'
+        else:
+            sai_threshold_value = fvs['static_th']
+            sai_threshold_mode = 'SAI_BUFFER_PROFILE_THRESHOLD_MODE_STATIC'
+        self.asic_db.wait_for_field_match("ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE", self.newProfileInAsicDb,
+                                             {'SAI_BUFFER_PROFILE_ATTR_XON_TH': fvs['xon'],
+                                              'SAI_BUFFER_PROFILE_ATTR_XOFF_TH': fvs['xoff'],
+                                              'SAI_BUFFER_PROFILE_ATTR_RESERVED_BUFFER_SIZE': fvs['size'],
+                                              'SAI_BUFFER_PROFILE_ATTR_POOL_ID': self.ingress_lossless_pool_oid,
+                                              'SAI_BUFFER_PROFILE_ATTR_THRESHOLD_MODE': sai_threshold_mode,
+                                              'SAI_BUFFER_PROFILE_ATTR_SHARED_DYNAMIC_TH': sai_threshold_value},
+                                          self.DEFAULT_POLLING_CONFIG)
+
+    def make_lossless_profile_name(self, speed, cable_length, mtu = None, dynamic_th = None):
+        extra = ""
+        if mtu:
+            extra += "_mtu" + mtu
+        if dynamic_th:
+            extra += "_th" + dynamic_th
+
+        return "pg_lossless_" + speed + "_" + cable_length + extra + "_profile"
 
-    @pytest.mark.skip(reason="Traditional buffer model runs on VS image currently.")
     def test_changeSpeed(self, dvs, testlog):
```

Comment:
check in asic db for this test case is missing

Reply:
We can’t do that because it very unlikely the old profile is removed after speed updated because the old speed is still being referenced by other ports, which means we can not know which oid is the new one